docs/CRT0.md

```diff
@@ -19,6 +19,54 @@ Only after this work has been done do we jump to the designated `main` symbol of
 This `main` must be able to be defined in either the guest language (and hence exposed in LLVM IR),
 or defined in Cairo itself.
 
+It looks something like this:
+
+```
+init:
+  __crt0_init()
+  __hieratika_main()
+```
+
+### Hieratika's Main Function
+
+As our platform is quite different to the operating systems and bare-metal platforms that are
+usually targeted by Rust code, which means that we can be more flexible with what we accept and
+return as part of it.
+
+Our main function is to be defined as a function with a signature that approximates the following.
+
+```rust
+#[no_mangle]
+fn hieratika_main(inputs: FeltArray) -> FeltArray {
+    // ...
+}
+```
+
+Note that `FeltArray` is currently a stand-in type for whatever type best approximates an LLVM
+`array` type, which will likely need to carry its length auxiliary to it. This could be something
+like LLVM's `[0 x i252]`—as zero-length arrays are fine in LLVM and just mean we can index
+arbitrarily into the memory represented by their pointer—combined with a `i32` for its length.
+
+Effectively, we want `Array` to be something similar to the following:
+
+```rust
+#[repr("C")]
+pub struct FeltArray {
+    /// A raw pointer to the first element in the backing array.
+    array: *mut Felt252,
+
+    /// The number of elements accessible through the pointer `array`.
+    length: usize
+}
+```
+
+In other words, it is a pointer to the first element of an array, and the number of elements in the
```

Comment:
In case `hieratika_main` is defined in Cairo there is no pointer arithmetic and it will use the type `Array<felt252>`. Should it be explicitly written?

Reply:
No, because that is not a supported use case. It has to be uniform for all uses.

Reply:
Line 20 above says "This `main` must be able to be [..] defined in Cairo itself"